### ISSUE TITLE
Switch From GitHub Commit Statuses API to Checks API

### ISSUE THREAD

**Issue by midnightconman** [kubernetes-sigs/prow#340]
Prow uses github commit statuses to interact with PRs (all the components that interact with these I don't know yet), should we switch from commit statuses to the new checks api? Checks seem much more featureful than commit statuses. Thoughts, comments, concerns?

**Comment by Prucek** [comment]
+1
I talked about this with @petr-muller, and he would agree that Prow deserves some modernization.

**Comment by petr-muller** [comment]
I'm favor of this generally. Two roadbumps I recall from earlier discussions, they may not describe the _current_ situation well though:

- Checks API had some issues with PAT authentications, IIRC parts of it were only available to GH Apps? I hope most Prow installations use GH Apps now but it is possible that we'd need to deprecate PATs
- IIRC check results were quite tricky to read / process from e.g. the Tide PoV, the check names / IDs were weird somehow. I remember this was causing issues when we tried to interface with GH actions better, but it's a while now

**Comment by midnightconman** [comment]
> Checks API had some issues with PAT authentications, IIRC parts of it were only available to GH Apps?

There is conflicting information in the github documents about this, where it says you need to use a github app to use various checks apis, but then it says it supports "Fine-grained personal access tokens": https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#create-a-check-run

**Comment by petr-muller** [comment]
Fine-grained personal access tokens are [quite new actually](https://github.blog/security/application-security/introducing-fine-grained-personal-access-tokens-for-github/), introduced in late 2022, which means they are newer than apps. My memory is definitely from before that. So it is quite likely that only the "old" PATs are not able to interact with checks.

**Comment by midnightconman** [comment]
So maybe we should think about deprecating "classic" personal access tokens usage?

**Comment by petr-muller** [comment]
Yeah, I would not be against that, but we'd need a reasonable transition period.
